syft/core/hooks/torch/hook.py

```diff
@@ -211,18 +216,194 @@ def function_router(*args, **kwargs):
+                    out.precision = self.precision
+                    out.fixed_precision = True
+
+                elif self.precision < other.precision:
+                    out = torch.LongTensor.old_add(self.old_mul(10 ** (other.precision - self.precision)), other)
+                    out.precision = other.precision
+                    out.fixed_precision = True
+
+                else:
+                    out = torch.LongTensor.old_add(self, other)
+                    out.precision = self.precision
+                    out.fixed_precision = True
+
+                return out
+            elif (hasattr(self, 'fixed_precision') and not hasattr(other, 'fixed_precision')) or \
+                 (not hasattr(self, 'fixed_precision') and hasattr(other, 'fixed_precision')):
+                raise AttributeError("Tried to call fixed precision operation on non-fixed precision tensor and "
+                                     "a regular tensor.")
+            else:
+                raise AttributeError("Tried to call fixed precision operation on non-fixed precision tensors")
+
+        tensor_type.fixed_prec_add = fixed_prec_add
+
+
+        def fixed_prec_sub(self, other):
+            if hasattr(self, 'fixed_precision') and hasattr(self, 'fixed_precision'):
+
+                if self.precision > other.precision:
+                    out = torch.LongTensor.old_sub(self, other.old_mul(10 ** (self.precision - other.precision)))
+                    out.precision = self.precision
+                    out.fixed_precision = True
+
+                elif self.precision < other.precision:
+                    out = torch.LongTensor.old_sub(self.old_mul(10 ** (other.precision - self.precision)), other)
+                    out.precision = other.precision
+                    out.fixed_precision = True
+
+                else:
+                    out = torch.LongTensor.old_sub(self, other)
+                    out.precision = self.precision
+                    out.fixed_precision = True
+
+                return out
+            elif (hasattr(self, 'fixed_precision') and not hasattr(other, 'fixed_precision')) or \
+                 (not hasattr(self, 'fixed_precision') and hasattr(other, 'fixed_precision')):
+                raise AttributeError("Tried to call fixed precision operation on non-fixed precision tensor and "
+                                     "a regular tensor.")
+            else:
+                raise AttributeError("Tried to call fixed precision operation on non-fixed precision tensors")
+
+        tensor_type.fixed_prec_sub = fixed_prec_sub
+
+        def fixed_prec_div(self, other):
+            if hasattr(self, 'fixed_precision') and hasattr(self, 'fixed_precision'):
+
+
+                if self.precision < other.precision:
+                    out = torch.LongTensor.old_div(self.old_mul(10 ** (other.precision - self.precision)), other)
+                    out.precision = 0
```

Comment:
Is it accurate for out.precision to always be 0 here? Seems like we would want it to be either the precision of self or other, no?

Reply:
since we have the same precision for both variables, out.precision should be 0 (10^0 = 1)
here is an example:
x = torch.FloatTensor([1,2,3,4,5]).set_precision(4)
y = torch.FloatTensor([1,2,3,4,5]).set_precision(3)
x = [10000,20000,30000,40000,50000]
y = [1000,2000,3000,4000,5000]

what the function does is first set the precision of y equal to x, so y is now [10000,20000,30000,40000,50000]. 

since they both have the same precision, dividing x by y gives us [1,1,1,1,1]. therefore, the precision of output is always 0 regardless of the either variable's precision, because of the conversion. 

hope that makes sense :)